src/plugin.js

```diff
@@ -1,122 +1,33 @@
-const fs = require('fs-extra');
-const glob = require('glob');
-const path = require('path');
-
-function getLocaleFiles() {
-  return glob.sync(
-    path.join('src/assets/locales', 'resources_*.json')
-  );
-}
-
-function readJson(file) {
-  fs.ensureFileSync(file);
-
-  const buffer = fs.readFileSync(file);
-
-  let contents;
-  // Is the locale file empty?
-  if (buffer.length === 0) {
-    contents = {};
-  } else {
-    contents = JSON.parse(buffer.toString());
-  }
-
-  return contents;
-}
-
-function getLocaleFromFileName(fileName) {
-  let locale = fileName.split('.json')[0].split('resources_')[1];
-
-  locale = locale.toUpperCase().replace('_', '-');
-
-  return locale;
-}
+const documentationGenerator = require('./documentation-generator');
+const resourcesProviderPlugin = require('./resources-provider');
+const sourceCodeProviderPlugin = require('./source-code-provider');
+const typeDocJsonProviderPlugin = require('./typedoc-json-provider');
 
 function SkyUXPlugin() {
-  const resourceFilesContents = {};
-  getLocaleFiles().forEach((file) => {
-    const locale = getLocaleFromFileName(file);
-    const contents = readJson(file);
-    resourceFilesContents[locale] = contents;
-  });
 
-  // Standardize keys to be uppercase, due to some language limitations
-  // with lowercase characters.
-  // See: https://stackoverflow.com/questions/234591/upper-vs-lower-case
-  const resourceFilesExist = ('EN-US' in resourceFilesContents);
-
-  const writeResourcesProvider = (content, resourcePath) => {
-    if (!resourceFilesExist) {
-      return content;
-    }
-
-    // Resolve the resource path for Windows machines.
-    const resolvedPath = path.resolve(resourcePath);
-
-    // Directory used when serving or building.
-    const dir = path.join('src', 'app', 'public', 'plugin-resources');
+  const preload = (content, resourcePath) => {
```

Comment:
Any reason this is still a function expression where most of the others in this PR are using the old school function statement?

Reply:
I tend to make functions within functions as "scopeless". (Arrow functions use the context of their wrapping object/function.)